spec/json_schema_spec.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true, encoding: ASCII-8BIT
+
+require File.expand_path("../support", __FILE__)
+
+class JsonSchemaBaseTest < CouchbaseOrm::Base
+  attribute :name, :string
+  attribute :numb, :integer
+
+  def initialize(model = nil, ignore_doc_type: nil, **attributes)
+    super
+    self.class.design_document('JsonSchemaBaseTest')
+  end
+end
+
+class UnknownTest < CouchbaseOrm::Base
+  attribute :test, :boolean
+end
+
+class EntitySnakecase < CouchbaseOrm::Base
+  attribute :value, :string
+end
+
+describe CouchbaseOrm::JsonSchema::Loader do
+
+  after(:each) do
+    CouchbaseOrm::JsonSchema::Loader.instance.reset
```

Comment:
We might accept to violate encapsulation in test code to prevent exposing methods for test purpose only.

```suggestion
    CouchbaseOrm::JsonSchema::Loader.instance.instance_variable_set(:@schemas, nil) 
```